org.eclipse.winery.edmm/src/main/java/org/eclipse/winery/edmm/utils/ZipUtility.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.winery.edmm.utils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class ZipUtility {
+
+    /**
+     * The function is used to zip a directory in a file
+     * @param sourcePath the path of the directory
+     * @param destPath the destination path of the zip file,
+     *                 if a zip file already exists at the destination path, it will be overwritten
+     *
+     * https://stackoverflow.com/questions/15968883/how-to-zip-a-folder-itself-using-java
+     */
+    public static ZipOutputStream pack(Path sourcePath, Path destPath) throws IOException {
```

Comment:
a non-obvious edge-case here is when `destPath.beginsWith(sourcePath)`, as in, the `destPath` is part of the tree that gets walked with `Files.walk`. In that case the code will probably deadlock without an error messge.

Reply:
If we want a general zip function we need to consider this case too.
 At the moment I think it won't happen a situation like this. When the function is called the `destPath` is never a sub tree of the `sourcePath`.